test/tests/functional/rs_mocks/read_preferences_tests.js

```diff
@@ -1337,4 +1337,104 @@ describe('ReplSet Read Preferences (mocks)', function() {
       });
     }
   });
+
```

Comment:
Can we please move this test to `tests/unit`, and use the `ReplSetFixture` to cut out a lot of the code dupe, and allow us to focus on the actual test conditions.